std/experimental/allocator/building_blocks/affix_allocator.d

```diff
@@ -186,11 +186,23 @@ struct AffixAllocator(Allocator, Prefix, Suffix = void)
 
         // Extra functions
         static if (stateSize!Prefix)
+        {
             static ref Prefix prefix(void[] b)
             {
                 assert(b.ptr && b.ptr.alignedAt(Prefix.alignof));
-                return (cast(Prefix*)b.ptr)[-1];
+                return (cast(Prefix*) b.ptr)[-1];
+            }
+            static ref shared(Prefix) prefix(const(void)[] b)
```

Comment:
Why not simply add a `shared` overload? I don't think that allowing a thread-local allocator's memory to be shared across threads is a good design. Only shared allocators (like `Mallocator.instance`) should allow shared use of a single memory allocation.